### DISCUSSION TITLE
Nextjs app running on Vercel recreates connections to Postgresql (Prisma)

### DISCUSSION THREAD

**Discussion by baunov** [vercel/community#3046] [Help]
### Problem Description

I'm currently hosting my Postgresql database on Aiven cloud. With NextJS app using Prisma running on Vercel. I noticed that Prisma client is reinitialised almost on each page navigation. And that causes page loading delays of up to 6 seconds which is definitely not acceptable. 
The queries in database by themselves take couple of milliseconds. Network latency also shouldn't be the case, I would accept like 1 second overall max. So, I suppose that those delays are caused by reinitilising Prisma and recreating database connections.
I logged out each time new PrismaClient is instantiated and that happens very often.

How to deal with that? And why does that happen? (is Vercel creating a new app instance on navigations?)

--
Also, I tried to cache in-memory on NextJS backend some entities that I fetch on every page load. But since the app is re-instantiated, I lose all that cached data.

**Comment by amyegan** [comment]
Hi @baunov. This sounds like very strange behavior. There might be something slightly off with the project's configuration. If you share your public repo or a minimal reproduction, then someone here might be able to recognize the problem
